Role update

ultraheroe opened the pull request:
### Description of the Change

Refactored template config files #4, fixed #5 and updated requirements, default versions and some default vars values.
.travis and .gitignore updated too

### Benefits

Maintains the role up to date with latest versions and (I wish) making the role fully configurable.

### Possible Drawbacks

Maybe not fully compatible with previous versions of this role

### Applicable Issues

#4
#5

Commit:
Merge pull request #3 from idealista/develop

Release 2.0.0

ultraheroe committed:
Updating role

- Defined hkp apt keyserver protocol
- Required vars for new config templates
- Fully parametrized config files
- .gitignore rules for various IDEs
- Deleted old stretch distro and add bullseye to .travis.yml
- Update test-requirements versions
- Update default version to latest (at this moment v21.11.2.2)
- Check how to check installed version
- Update molecule config and vars

ultraheroe committed:
Rename vars because wrong naming

- It's not okay to use camelCase in ansible vars
- renamed in necessary files

ultraheroe committed:
Update Changelog

ultraheroe committed:
Update .travis dist

ablancolopez commented on molecule/default/Dockerfile.j2:
```diff
@@ -8,5 +8,5 @@ FROM {{ item.image }}
 
 # install minimal packages for debian slim images
 RUN apt-get update && \
```
Be aware that this would probably break the build in Debian Buster, as the Python package is still being called `python`. See Prometheus Node Exporter role configuration for reference: https://github.com/idealista/prometheus_node_exporter_role/blob/develop/molecule/default/Dockerfile.j2

ultraheroe replied:
Roger, fixed now

ablancolopez commented on molecule/default/molecule.yml:
```diff
@@ -23,15 +23,27 @@ platforms:
       - /run/lock
     volumes:
       - '/sys/fs/cgroup:/sys/fs/cgroup:ro'
+      - '/home/david/Desktop/ch:/var/lib/clickhouse/'
```
I think this is heavily dependant on your personal environment :P

ultraheroe replied:
done!

ultraheroe committed:
Update molecule.yml

- Remove personal volume (ups!)
- Add requiered exposed port for jdbc-bridge

ultraheroe committed:
Updatge dockerfile for debian11 and 10 support

ultraheroe committed:
Fix and update templates

- Fix clickhouse_limit_no_file in templates
- Add custom vars for custom config in config files

ablancolopez commented on defaults/main.yml:
```diff
@@ -30,7 +30,7 @@ clickhouse_service_state: started
 
 clickhouse_private_tmp: false
-clickhouse_limit_NO_FILE: 500000
+clickhouse_limit_no_file: 500000
```
Personally I would rather keep the old naming to be consistent with Ansible naming conventions

ablancolopez replied:
wow, i misread, marking as resolved :)

ultraheroe committed:
Update tests

- Update users grants tests

ultraheroe committed:
Update README.md
